__main__.py

```diff
@@ -0,0 +1,211 @@
+        'code': 'CLJ_Code'
+    }, inplace=True)
+
+    clojure_master["CLJ"] = '+'
+
+    clojure_master.drop(columns = ['offset'], inplace = True)
+    # display(clojure_master)
+
+
+    # In[5]:
+
+
+    # Merge Clojure and CodeQL dataframes
+
+    for key, df in dataframes.items():
+        k = pd.merge(clojure_master[clojure_master["Type"] == key], df, on = ['File', 'Type', 'Line'], how = 'outer')
+        k = k[["Type", "File", "Line", "CLJ", "CQL", "CLJ_Code", "CQL_Column", "CQL_Code"]]
+        k = k.drop_duplicates(subset = ["Type", "File", "Line", "CLJ", "CQL"], keep = 'first')
+        k['CLJ'] = k['CLJ'].fillna('-')
+        k['CQL'] = k['CQL'].fillna('-')
+        dataframes[key] = k.sort_values(by = ["File", "Line"]).reset_index().drop("index", axis = 1)
+        print(f"Merged DataFrame for {key}:")
+    #     display(dataframes[key].head())
+
+
+    # In[6]:
+
+
+    # Load Coccinelle dataframes
+
+    coccinelle_dataframes = {}
+    for filename in os.listdir(coccinelle_directory):
+        if filename.endswith('.csv'):
+            file_path = os.path.join(coccinelle_directory, filename)
+            dataframe_key = filename[:-4]
+            p = pd.read_csv(file_path)
+            p.columns = ["Type", "File", "Line", "CNL_Column", "CNL_Code"]
+            p["File"] = p["File"].str.split('/').str[-1]
+            p["CNL"] = '+'
+            coccinelle_dataframes[dataframe_key] = p
+    print(coccinelle_dataframes.keys())
+
+    coccinelle_mapping = {
+        'post-increment': 'postIncr',
+        'assignment-as-value': 'assignment_as_value',
+        'conditional' : 'conditional',
+        'pre-increment': 'preIncr',
+        'comma-operator': 'comma_atoms',
+    }
+
+
+    for i in coccinelle_dataframes.values():
+        i['Type'] = i['Type'].map(coccinelle_mapping)
+    #     display(i)
+
+
+    # In[7]:
+
+
+    # Merge Coccinelle DataFrames
+
+    for i in coccinelle_dataframes.values():
+        key = i.loc[0, "Type"]
+        toMerge = dataframes[key]
+        k = pd.merge(toMerge, i, on = ['File', 'Type', 'Line'], how = 'outer')
+        k['CLJ'] = k['CLJ'].fillna('-')
+        k['CQL'] = k['CQL'].fillna('-')
+        k['CNL'] = k['CNL'].fillna('-')
+        # Reordering columns
+        k = k[["Type", "File", "Line", "CLJ", "CQL", "CNL", "CLJ_Code", "CQL_Column", "CQL_Code", "CNL_Column", "CNL_Code"]]
+        k = k.drop_duplicates(subset = ["Type", "File", "Line", "CLJ", "CQL", "CNL"], keep = 'first')
+        # Float type For consistency with CQL_Column, which is forced as a float because of NaN values
+        k["CQL_Column"] = k["CQL_Column"].astype(float)
+        dataframes[key] = k.sort_values(by = ["File", "Line"]).reset_index().drop("index", axis = 1)
+        print(f"Merged DataFrame for {key}:")
+    #     display(dataframes[key].head())
+
+
+    # In[8]:
+
+
+    # Output merged CSVs to output_directory
+
+    if (output_directory!="./output") and (not os.path.exists(output_directory)):
```

Comment:
I do not advise directly comparing paths in such a way, due to the differences between Windows and Unix paths. I wholeheartedly recommend using `pathlib`. If `output_directory` is a `Path` instance, just check `noy output_directory.is_dir()`. No reason to compare with `output`. If `output` exists already, it will not be created because of this check. You can also use `exist_ok` when creating the dir, in which case you don't even have to check if it exists